### ISSUE TITLE
Want to open two AIRSPY HF+ at the same time

### ISSUE THREAD

**Issue by n1mm** [IW0HDV/extio-iw0hdv#14]
I have an application that interfaces with Airspy HF+ via your extio.  My users use the SDRs as a panadapter adjunct to their ham radio.  Some of the users operate with two ham radios at a time in contests and would like to be able to configure two HF+'s at a time. 

I tried setting the environment variable after opening the first instance of your extio, but setting it from within my program (vb.net) did not work.  If I set the environment variable (AIRSPYHF) externally, prior to starting my program, your extio picks up the serial #, but then both instances of your extio try to use the same hardware. 

What would be ideal (for me) would be for your extio to open one and then if a second instance is opened, for it to sense that the first hardware was in use and then default to opening the second HF+.  

Any suggestions for a solution would be appreciated.

73,
Tom - N1MM
https://n1mmwp.hamdocs.com/

**Comment by amontefusco** [comment]
In order to use two different hardware, you need two main processes.

 Andrea Montefusco IW0HDV

**Comment by amontefusco** [comment]
The ExtIO API is modeled on a very simple 1:1 relationship: for each SDR program running, you can manage just one piece of hardware.
In order to use two radios at the same time, you have to run two instances of SDR software (e.g. you have to start HDSDR twice, each one with the right environment or command line able to select the wanted hardware instance).

So, in your case, you could manage to start two (or more) subprocesses, each one managing its radio. Processes not threads.

**Comment by n1mm** [comment]
Thanks for looking at the request.  I already have multiple SDRs supported via extio in threads and don't want to change that.  I also support spyserver and that is how this user is handling the issue now.  He can specify the serial # in spyserver, and use the environment variable to force the one supported via your extio to the other one.  

I may look at downloading your source and add the feature myself, but I've done little work with Github builds, so it will be a steep learning curve.  I will probably wait for more users to ask for the capability before devoting time to that. 

If you would like to see how I am implementing this interface, this video will illustrate.  The main part of the video starts at 0:30.
https://www.youtube.com/watch?v=nZUKbKJo4P4

73 - Tom - N1MM

**Comment by touil** [comment]
Hello guys. How about using the C# wrapper and call it from VB.NET?

**Comment by n1mm** [comment]
We already have an extio interface that connects to multiple vendors SDRs.  Kinda like to stick with that for now.  I looked at your C# wrapper, but that is more work than modifying this extio interface to find an alternate Airspy HF+.  As I implied above, I only know of one user with two Airspy's so far, so that's a lot of time that could be spent elsewhere. 

The spyserver interface is getting kudos for letting ops remote their SDR and display the spectrum at home with reasonable bandwidth.
